iree/compiler/Dialect/Flow/Transforms/DispatchConfig.cpp

```diff
@@ -48,8 +48,26 @@ bool isIndexOp(Operation *op) {
          isa<mhlo::BroadcastInDimOp>(op) || isa<mhlo::BroadcastOp>(op) ||
          isa<mhlo::DynamicBroadcastInDimOp>(op) ||
          isa<mhlo::DynamicReshapeOp>(op) || isa<mhlo::DynamicSliceOp>(op) ||
-         isa<mhlo::ReshapeOp>(op) || isa<mhlo::SliceOp>(op) ||
-         isa<mhlo::TransposeOp>(op);
+         isa<mhlo::SliceOp>(op) || isa<mhlo::TransposeOp>(op);
+}
+
+/// Returns true if |lhs| and |rhs| return a single value with the same shape.
+bool checkSameOutputShape(Operation *lhs, Operation *rhs) {
+  if (lhs->getNumResults() != 1 || rhs->getNumResults() != 1) {
+    return false;
+  }
+  ShapedType lhsType = lhs->getResults()[0].getType().dyn_cast<ShapedType>();
+  ShapedType rhsType = rhs->getResults()[0].getType().dyn_cast<ShapedType>();
+  if (!lhsType || !rhsType || lhsType != rhsType) {
+    return false;
+  }
```

Comment:
[optional] `if (!lhsType || lhsType != rhsType)` is enough, but I think maybe the current way is easier to understand?

Reply:
True. But as you said, this is easier to understand the condition,